vrecord

```diff
@@ -957,25 +946,28 @@ else
 fi
 
 if [[ "${video_bit_depth_choice}" && "${video_bit_depth_choice}" != "${undeclaredoption}" ]] ; then
-    lookup_video_bitdepth "${video_bit_depth_choice}"
+    lookup_pixel_format "${video_bit_depth_choice}"
 else
     _report -q "Which VIDEO bit depth?"
     PS3="Select a video bit depth: "
-    select video_bit_depth_choice in "${video_bitdepth_options[@]}" ; do
-        lookup_video_bitdepth "$video_bit_depth_choice"
+    select video_bit_depth_choice in "${pixel_format_options[@]}" ; do
+        lookup_pixel_format "$video_bit_depth_choice"
         [ "${?}" -eq 0 ] && break
     done
 fi
 
-GRAB_DECKLINK=(bmdcapture)
-GRAB_DECKLINK+=(-m ${standard})
-GRAB_DECKLINK+=(-V "${video_input}")
-GRAB_DECKLINK+=(-A "${audio_input}")
-GRAB_DECKLINK+=(-p "${video_bitdepth}")
-GRAB_DECKLINK+=(-c 8)
-GRAB_DECKLINK+=(-s 32)
-GRAB_DECKLINK+=(-F nut)
-GRAB_DECKLINK+=(-f pipe:1)
+GRAB_DECKLINK=(ffmpeg-dl)
+GRAB_DECKLINK+=(-f decklink)
+GRAB_DECKLINK+=(-audio_input "${audio_input}")
+GRAB_DECKLINK+=(-video_input "${video_input}")
+GRAB_DECKLINK+=(-format_code "${standard}")
+GRAB_DECKLINK+=(-channels 8)
+GRAB_DECKLINK+=(-raw_format "${pixel_format}")
+GRAB_DECKLINK+=(-i 'UltraStudio Express')
```

Comment:
Is this a hardcoded value?what if you had another decklink or ultrqstudio card?

Reply:
I didn’t get around to writing a query for the devices yet.

Reply:
cool